File: java/utility/PreparePath.java
package voxgig.hook0sdk.utility;

import java.util.ArrayList;
import java.util.List;

import voxgig.hook0sdk.core.Context;
import voxgig.hook0sdk.utility.struct.Struct;

@SuppressWarnings({"unchecked"})
final class PreparePath {

  private PreparePath() {}

  static String preparePath(Context ctx) {
    List<Object> parts = null;
    Object p = Struct.getprop(ctx.point, "parts");
    if (p instanceof List) {
      parts = (List<Object>) p;
    }
    if (parts == null) {
      parts = new ArrayList<>();
    }

    return Struct.join(parts, "/", true);
  }
}

File: .sdk/tm/csharp/utility/PreparePath.cs
// Hook0 SDK utility: preparePath.

using Voxgig.Struct;

namespace Hook0Sdk.Util;

public static partial class SdkUtility
{
    internal static string PreparePathUtil(Context ctx)
    {
        var point = ctx.Point;

        var parts = StructUtils.GetProp(point, "parts") as List<object?>
            ?? new List<object?>();

        return StructUtils.Join(parts, "/", true);
    }
}
